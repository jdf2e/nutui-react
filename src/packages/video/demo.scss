.demo-video {
  padding: 57px 0 0 0 !important;
}
.video-cell {
  padding: 0;
  &::after {
    display: none;
  }
}
.video-m-b {
  margin-bottom: 60px;
}
