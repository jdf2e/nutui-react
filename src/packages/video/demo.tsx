import React from 'react'
import { useTranslate } from '../../sites/assets/locale'

import Demo1 from './demos/h5/demo1'
import Demo2 from './demos/h5/demo2'
import Demo3 from './demos/h5/demo3'
import Demo4 from './demos/h5/demo4'
import Demo5 from './demos/h5/demo5'
import Demo6 from './demos/h5/demo6'
import Demo7 from './demos/h5/demo7'

const VideoDemo = () => {
  const [translated] = useTranslate({
    'zh-CN': {
      basic: '基础用法',
      autoPlay: '自动播放',
      muted: '初始化静音',
      cover: '视频封面海报设置',
      inline: '行内播放',
      background: '设置视频为背景图',
      switch: '视频切换',
    },
    'zh-TW': {
      basic: '基礎用法',
      autoPlay: '自動播放',
      muted: '初始化靜音',
      cover: '視頻封面海報設置',
      inline: '行內播放',
      background: '設置視頻為背景圖',
      switch: '視頻切換',
    },
    'en-US': {
      basic: 'Basic Usage',
      autoPlay: 'Auto play',
      muted: 'Initialize mute',
      cover: 'Video cover poster settings',
      inline: 'play inline',
      background: 'Set video as background',
      switch: 'Video switching',
    },
  })
  return (
    <>
      <div className="demo full">
        <h2>{translated.basic}</h2>
        <Demo1 />

        <h2>{translated.autoPlay}</h2>
        <Demo2 />

        <h2>{translated.muted}</h2>
        <Demo3 />

        <h2>{translated.cover}</h2>
        <Demo4 />

        <h2>{translated.inline}</h2>
        <Demo5 />

        <h2>{translated.background}</h2>
        <Demo6 />

        <h2>{translated.switch}</h2>
        <Demo7 />
      </div>
    </>
  )
}

export default VideoDemo
