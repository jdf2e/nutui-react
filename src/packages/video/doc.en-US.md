#  Video 

### Intro

Video player implemented by native video

### Install

``` ts
import { Video } from '@nutui/nutui-react';
```

### Basic Usage

:::demo
```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>Basic Usage</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlayFuc={play}
          onPauseFuc={pause}
          onPlayend={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```
:::

### Auto play
autoplay Property to set video autoplay

:::demo
```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    autoplay: true,
    muted: true,
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>Auto play</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlayFuc={play}
          onPauseFuc={pause}
          onPlayend={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```
:::

### Initialize mute
The muted property sets the initial mute of the video

:::demo
```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    muted: true,
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>Initialize mute</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlayFuc={play}
          onPauseFuc={pause}
          onPlayend={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```
:::

### Video cover poster settings
The poster property sets the video poster

:::demo
```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
    poster:
      'https://img12.360buyimg.com/ling/s345x208_jfs/t1/168105/33/8417/54825/603df06dEfcddc4cb/21f9f5d0a1b3dad4.jpg.webp',
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>Video cover poster settings</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlayFuc={play}
          onPauseFuc={pause}
          onPlayend={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```

:::

### play inline
The playsinline property sets the mobile terminal video to play in line and prevents the newly opened page from playing (compatible with IOS and some Android machines)

:::demo
```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
    playsinline: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>play inline</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlayFuc={play}
          onPauseFuc={pause}
          onPlayend={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```
:::

### Set video as background
When setting the video as the background image, it is necessary to set muted, disabled, operation prohibited, loop, loop and autoplay to true, and the mobile terminal needs to set playinline for in-line display

:::demo
```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: false,
    autoplay: true,
    disabled: true,
    muted: true,
    playsinline: true,
    loop: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>Set video as background</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlayFuc={play}
          onPauseFuc={pause}
          onPlayend={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```
:::

### Video switching
Reset the video when the video address changes

:::demo
```tsx
import React, { useState } from "react";
import { Cell, Video, Button } from '@nutui/nutui-react';

const App = () => {
  const [source1, setSource1] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/legao-%E6%9D%A8%E8%BF%9B%E5%86%9B.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)

  const changeVideo = () => {
    setSource1({...source1, src: 'https://vjs.zencdn.net/v/oceans.mp4'})
  }
  return (
    <>
      <h2>Video switching</h2>
      <Cell className='cell'>
        <Video
          source={source1}
          options={options}
          onPlayFuc={play}
          onPauseFuc={pause}
          onPlayend={playend}
         />
      </Cell>
      <Button type="primary" className="m-b" onClick={changeVideo}>Video switching</Button>
    </>
  )
}
export default App;
```
:::


## API

### Props

| Attribute                | Description                                       | Type    | Default   |
| ------------------- | ------------------------------------------ | ------- | -------- |
| source             | Video url and type settings                         | Object   | -        |
| options             | Control video playback properties                          | Object  | required |
| options.autoplay    | Auto play                               | Boolean | false    |
| options.poster      | Poster settings                                | String  | -        |
| options.loop        | Poster loop                             | Boolean | false    |
| options.controls    | Show operation control                             | Boolean | true     |
| options.muted       | Mute                                   | Boolean | false    |
| options.playsinline | Whether to set as inline playback element (solve Android compatibility problem) | Boolean | false    |

### Events

| Event | Description	         | Arguments |
| -------- | ------------ | -------- |
| play `v1.3.8 Abandon`     | play event         | --       |
| pause `v1.3.8 Abandon`    | pause event         | --       |
| playend `v1.3.8 Abandon`  | Playback completion callback | --       |
| onPlayFuc `v1.3.8`     | play event         | --       |
| onPauseFuc `v1.3.8`    | pause event         | --       |
| onPlayend `v1.3.8`  | Playback completion callback | --       |
