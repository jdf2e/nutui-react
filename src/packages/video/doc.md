# Video 视频播放器

## 介绍

原生video实现的视频播放器

## 安装

```ts
import { Video } from '@nutui/nutui-react';
```

## 代码演示

### 基础用法

:::demo

```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>基础用法</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlay={play}
          onPause={pause}
          onPlayEnd={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```

:::

### 自动播放

autoplay 属性设置视频自动播放

:::demo

```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    autoplay: true,
    muted: true,
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>自动播放</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlay={play}
          onPause={pause}
          onPlayEnd={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```

:::

### 初始化静音

muted属性设置视频初始化静音

:::demo

```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    muted: true,
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>初始化静音</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlay={play}
          onPause={pause}
          onPlayEnd={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```

:::

### 视频封面海报设置

poster 属性设置视频海报

:::demo

```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
    poster:
      'https://img12.360buyimg.com/ling/s345x208_jfs/t1/168105/33/8417/54825/603df06dEfcddc4cb/21f9f5d0a1b3dad4.jpg.webp',
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>视频封面海报设置</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlay={play}
          onPause={pause}
          onPlayEnd={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```

:::

### 行内播放

playsinline 属性设置移动端视频行内播放，阻止新打开页面播放（兼容 ios，兼容部分安卓机）

:::demo

```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
    playsinline: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>行内播放</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlay={play}
          onPause={pause}
          onPlayEnd={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```

:::

### 视频背景图

当设置视频为背景图时需要将 muted 静音、 disabled 禁止操作、loop 循环播放、autoplay 自动播放设置为 true，移动端需要设置 playsinline 行内展示

:::demo

```tsx
import React, { useState } from "react";
import { Cell, Video } from '@nutui/nutui-react';

const App = () => {
  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: false,
    autoplay: true,
    disabled: true,
    muted: true,
    playsinline: true,
    loop: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)
  return (
    <>
      <h2>设置视频为背景图</h2>
      <Cell className='cell'>
        <Video
          source={source}
          options={options}
          onPlay={play}
          onPause={pause}
          onPlayEnd={playend}
         />
      </Cell>
    </>
  )
}
export default App;
```

:::

### 视频切换

当视频地址发生变化时，重置视频

:::demo

```tsx
import React, { useState } from "react";
import { Cell, Video, Button } from '@nutui/nutui-react';

const App = () => {
  const [source1, setSource1] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/legao-%E6%9D%A8%E8%BF%9B%E5%86%9B.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)

  const changeVideo = () => {
    setSource1({...source1, src: 'https://vjs.zencdn.net/v/oceans.mp4'})
  }
  return (
    <>
      <h2>视频切换</h2>
      <Cell className='cell'>
        <Video
          source={source1}
          options={options}
          onPlay={play}
          onPause={pause}
          onPlayEnd={playend}
         />
      </Cell>
      <Button type="primary" className="m-b" onClick={changeVideo}>视频切换</Button>
    </>
  )
}
export default App;
```

:::

## Video

### Props

| 属性 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| source | 视频地址和类型设置 | `object` | `{type: {}, src: ''}` |
| options | 控制视频播放属性 | `object` | `-` |
| options.autoplay | 是否自动播放 | `boolean` | `false` |
| options.poster | 海报设置 | `string` | `-` |
| options.loop | 是否循环播放 | `boolean` | `false` |
| options.controls | 是否展示操作栏 | `boolean` | `true` |
| options.muted | 是否静音 | `boolean` | `false` |
| options.playsinline | 是否设置为行内播放元素（解决安卓兼容问题） | `boolean` | `false` |
| onPlay | 播放 | `(element: HTMLVideoElement) => void` | `-` |
| onPause | 暂停 | `(element: HTMLVideoElement) => void` | `-` |
| onPlayEnd | 播放完成回调 | `(element: HTMLVideoElement) => void` | `-` |