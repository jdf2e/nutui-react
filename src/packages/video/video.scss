.nut-video {
  width: 100%;
  height: 100%;
  position: relative;
  display: flex;

  &-player {
    width: 100%;
    background: #000;

    &:focus {
      outline: none;
    }
  }

  video {
    width: 100%;
    height: 100%;
    object-fit: fill;
  }
}
