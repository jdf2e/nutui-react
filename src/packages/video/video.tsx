import React, { useEffect, useRef, FunctionComponent } from 'react'
import classNames from 'classnames'
import bem from '@/utils/bem'

export interface VideoProps {
  source: {
    type: string
    src: string
  }
  options: {
    controls?: boolean
    muted?: boolean
    autoplay?: boolean
    poster?: string
    playsinline?: boolean
    loop?: boolean
  }
  className: string
  style: React.CSSProperties
  play: (e: HTMLVideoElement) => void
  pause: (e: HTMLVideoElement) => void
  playend: (e: HTMLVideoElement) => void
  onPlayFuc: (e: HTMLVideoElement) => void
  onPauseFuc: (e: HTMLVideoElement) => void
  onPlayend: (e: HTMLVideoElement) => void
}
const defaultProps = {
  source: {
    type: {},
    src: '',
  },
  options: {
    controls: true,
    muted: false, // 默认不是静音
    autoplay: false,
    poster: '',
    playsinline: false,
    loop: false,
  },
} as VideoProps
export const Video: FunctionComponent<
  Partial<VideoProps> & React.HTMLAttributes<HTMLDivElement>
> = (props) => {
  const {
    children,
    source,
    options,
    className,
    play,
    pause,
    playend,
    onPlayFuc,
    onPauseFuc,
    onPlayend,
    ...restProps
  } = {
    ...defaultProps,
    ...props,
  }
  const rootRef = useRef<HTMLVideoElement>(null)
  const b = bem('video')
  const classes = classNames(className, b(''))

  useEffect(() => {
    init()
  }, [])

  const init = () => {
    if (rootRef.current) {
      const videoRef = rootRef.current
      if (options.autoplay) {
        setTimeout(() => {
          videoRef.play()
        }, 200)
      }
      if (options.playsinline) {
        videoRef.setAttribute('playsinline', String(options.playsinline))
        videoRef.setAttribute('webkit-playsinline', String(options.playsinline))
        videoRef.setAttribute('x5-video-player-type', 'h5-page')
        videoRef.setAttribute('x5-video-player-fullscreen', 'false')
      }
      videoRef.addEventListener('play', () => {
        onPlayFuc && onPlayFuc(videoRef)
        play && play(videoRef)
      })
      videoRef.addEventListener('pause', () => {
        onPauseFuc && onPauseFuc(videoRef)
        pause && pause(videoRef)
      })
      videoRef.addEventListener('ended', () => {
        videoRef.currentTime = 0
        onPlayend && onPlayend(videoRef)
        playend && playend(videoRef)
      })
    }
  }

  return (
    <div className={classes} {...restProps}>
      <video
        className="nut-video-player"
        muted={options.muted}
        autoPlay={options.autoplay}
        loop={options.loop}
        poster={options.poster}
        controls={options.controls}
        ref={rootRef}
        src={source.src}
      >
        <source src={source.src} type={source.type} />
        <track kind="captions" />
      </video>
    </div>
  )
}

Video.defaultProps = defaultProps
Video.displayName = 'NutVideo'
