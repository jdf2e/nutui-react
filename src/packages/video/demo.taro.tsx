import React, { useState } from 'react'
import { useTranslate } from '@/sites/assets/locale/taro'
import { Cell, Button, Video } from '@/packages/nutui.react.taro'
import Header from '@/sites/components/header'
import Taro from '@tarojs/taro'
import '@/packages/video/demo.scss'

interface T {
  '84aa6bce': string
  a5a25e88: string
  '19875a3f': string
  fcdac2ed: string
  '200baa8c': string
  '5ec0e561': string
  a6e0b0cf: string
}
const VideoDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      '84aa6bce': '基础用法',
      a5a25e88: '自动播放',
      '19875a3f': '初始化静音',
      fcdac2ed: '视频封面海报设置',
      '200baa8c': '行内播放',
      '5ec0e561': '设置视频为背景图',
      a6e0b0cf: '视频切换',
    },
    'zh-TW': {
      '84aa6bce': '基礎用法',
      a5a25e88: '自動播放',
      '19875a3f': '初始化靜音',
      fcdac2ed: '視頻封面海報設置',
      '200baa8c': '行內播放',
      '5ec0e561': '設置視頻為背景圖',
      a6e0b0cf: '視頻切換',
    },
    'en-US': {
      '84aa6bce': 'Basic Usage',
      a5a25e88: 'Auto play',
      '19875a3f': 'Initialize mute',
      fcdac2ed: 'Video cover poster settings',
      '200baa8c': 'play inline',
      '5ec0e561': 'Set video as background',
      a6e0b0cf: 'Video switching',
    },
  })

  const [source, setSource] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/video_NutUI.mp4',
    type: 'video/mp4',
  })
  const [source1, setSource1] = useState({
    src: 'https://storage.360buyimg.com/nutui/video/legao-%E6%9D%A8%E8%BF%9B%E5%86%9B.mp4',
    type: 'video/mp4',
  })
  const options = {
    controls: true,
  }
  const options1 = {
    autoplay: true,
    muted: true,
    controls: true,
  }
  const options2 = {
    muted: true,
    controls: true,
  }
  const options3 = {
    controls: true,
    poster:
      'https://img12.360buyimg.com/ling/s345x208_jfs/t1/168105/33/8417/54825/603df06dEfcddc4cb/21f9f5d0a1b3dad4.jpg.webp',
  }
  const options4 = {
    controls: true,
    playsinline: true,
  }
  const options5 = {
    controls: false,
    autoplay: true,
    disabled: true,
    muted: true,
    playsinline: true,
    loop: true,
  }
  const play = (elm: HTMLVideoElement) => console.log('play', elm)
  const pause = (elm: HTMLVideoElement) => console.log('pause', elm)
  const playend = (elm: HTMLVideoElement) => console.log('playend', elm)

  const changeVideo = () => {
    setSource1({ ...source1, src: 'https://vjs.zencdn.net/v/oceans.mp4' })
  }
  return (
    <>
      <Header />
      <div
        className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''} demo-video`}
      >
        <h2>{translated['84aa6bce']}</h2>
        <Cell className="video-cell">
          <Video
            source={source}
            options={options}
            onPlayFuc={play}
            onPauseFuc={pause}
            onPlayend={playend}
          />
        </Cell>
        <h2>{translated.a5a25e88}</h2>
        <Cell className="video-cell">
          <Video
            source={source}
            options={options1}
            onPlayFuc={play}
            onPauseFuc={pause}
            onPlayend={playend}
          />
        </Cell>
        <h2>{translated['19875a3f']}</h2>
        <Cell className="video-cell">
          <Video
            source={source}
            options={options2}
            onPlayFuc={play}
            onPauseFuc={pause}
            onPlayend={playend}
          />
        </Cell>
        <h2>{translated.fcdac2ed}</h2>
        <Cell className="video-cell">
          <Video
            source={source}
            options={options3}
            onPlayFuc={play}
            onPauseFuc={pause}
            onPlayend={playend}
          />
        </Cell>
        <h2>{translated['200baa8c']}</h2>
        <Cell className="video-cell">
          <Video
            source={source}
            options={options4}
            onPlayFuc={play}
            onPauseFuc={pause}
            onPlayend={playend}
          />
        </Cell>
        <h2>{translated['5ec0e561']}</h2>
        <Cell className="video-cell">
          <Video
            source={source}
            options={options5}
            onPlayFuc={play}
            onPauseFuc={pause}
            onPlayend={playend}
          />
        </Cell>
        <h2>{translated.a6e0b0cf}</h2>
        <Cell className="video-cell">
          <Video
            source={source1}
            options={options}
            onPlayFuc={play}
            onPauseFuc={pause}
            onPlayend={playend}
          />
        </Cell>
        <Button type="primary" className="video-m-b" onClick={changeVideo}>
          {translated.a6e0b0cf}
        </Button>
      </div>
    </>
  )
}

export default VideoDemo
