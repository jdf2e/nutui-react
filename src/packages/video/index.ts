import { Video } from './video'

export type { VideoProps } from './video'
export default Video
