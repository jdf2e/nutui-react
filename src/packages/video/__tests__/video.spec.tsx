import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { Video } from '../video'

test('video base info', () => {
  const App = () => {
    const source = {
      src: 'xxx.mp4',
      type: 'video/mp4',
    }
    const options = {
      controls: true,
      autoplay: true,
      playsinline: true,
      loop: true,
      poster:
        'https://img12.360buyimg.com/ling/s345x208_jfs/t1/168105/33/8417/54825/603df06dEfcddc4cb/21f9f5d0a1b3dad4.jpg.webp',
    }
    return <Video source={source} options={options} />
  }

  const { container } = render(<App />)
  const sourceDom = container.querySelector('.nut-video source')
  expect(sourceDom?.getAttribute('src')).toBe('xxx.mp4')
  expect(container).toMatchSnapshot()
})
