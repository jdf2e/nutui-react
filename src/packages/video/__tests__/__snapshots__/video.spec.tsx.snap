// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`video base info 1`] = `
<div>
  <div
    class="nut-video"
  >
    <video
      autoplay=""
      class="nut-video-player"
      controls=""
      loop=""
      poster="https://img12.360buyimg.com/ling/s345x208_jfs/t1/168105/33/8417/54825/603df06dEfcddc4cb/21f9f5d0a1b3dad4.jpg.webp"
      src="xxx.mp4"
    >
      <source
        src="xxx.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;
